account_bank_statement_import_paypal/models/account_bank_statement_import_paypal_parser.py

```diff
@@ -70,93 +70,101 @@ def parse(self, mapping, data_file, filename):
+                data_dict[key] = None
+        return data_dict
+
     def _parse_lines(self, mapping, data_file, currency_code):
         data = StringIO(data_file.decode("utf-8-sig"))
         csv_data = reader(data)
 
         header = list(next(csv_data))
-        date_column = header.index(mapping.date_column)
-        time_column = header.index(mapping.time_column)
-        tz_column = header.index(mapping.tz_column)
-        name_column = header.index(mapping.name_column)
-        currency_column = header.index(mapping.currency_column)
-        gross_column = header.index(mapping.gross_column)
-        fee_column = header.index(mapping.fee_column)
-        balance_column = header.index(mapping.balance_column)
-        transaction_id_column = header.index(mapping.transaction_id_column)
-        try:
-            description_column = header.index(mapping.description_column)
-        except ValueError:
-            description_column = None
-        try:
-            type_column = header.index(mapping.type_column)
-        except ValueError:
-            type_column = None
-        try:
-            from_email_address_column = header.index(mapping.from_email_address_column)
-        except ValueError:
-            from_email_address_column = None
-        try:
-            to_email_address_column = header.index(mapping.to_email_address_column)
-        except ValueError:
-            to_email_address_column = None
-        try:
-            invoice_id_column = header.index(mapping.invoice_id_column)
-        except ValueError:
-            invoice_id_column = None
-        try:
-            subject_column = header.index(mapping.subject_column)
-        except ValueError:
-            subject_column = None
-        try:
-            note_column = header.index(mapping.note_column)
-        except ValueError:
-            note_column = None
-        try:
-            bank_name_column = header.index(mapping.bank_name_column)
-        except ValueError:
-            bank_name_column = None
-        try:
-            bank_account_column = header.index(mapping.bank_account_column)
-        except ValueError:
-            bank_account_column = None
+        data_dict = self._data_dict_constructor(mapping, header)
+
+        return self._calculate_lines(csv_data, data_dict, mapping, currency_code)
 
+    def _calculate_lines(self, csv_data, data_dict, mapping, currency_code):
         lines = []
         for row in csv_data:
             row = list(row)
-            date_value = row[date_column]
-            time_value = row[time_column]
-            tz_value = row[tz_column]
-            name_value = row[name_column]
-            currency_value = row[currency_column]
-            gross_value = row[gross_column]
-            fee_value = row[fee_column]
-            balance_value = row[balance_column]
-            transaction_id_value = row[transaction_id_column]
+            date_value = row[data_dict.get("date_column")]
```

Comment:
No need to use `.get` if the value has already fetched or filled with `None`, isn't it?